OpenScienceJournal/whistlepunk_library/src/main/java/com/google/android/apps/forscience/whistlepunk/ExportService.java

```diff
@@ -145,7 +145,11 @@ public static void exportTrial(Context context, String experimentId, String tria
     public static void cleanOldFiles(Context context) {
         Intent intent = new Intent(context, ExportService.class);
         intent.setAction(ACTION_CLEAN_OLD_FILES);
-        context.startService(intent);
+        try {
+            context.startService(intent);
+        } catch (Throwable ignored) {
```

Comment:
Catching Throwable is generally not a good idea. Can you explain why this is needed?

Reply:
After I raised the build target to Android 9, it happened that the app crashed because that service (called within the cleanOldFiles() method) could be started while the app was in background mode (and started service is not a foreground service). That general try ... catch avoids app crashing, and in that case the cleanOldFiles() will be reattempted when the app comes back in foreground.

Reply:
Would catching Exception still work?